presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusTable.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.connector.ColumnMetadata;
+
+import java.net.URI;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+public class PrometheusTable
+{
+    private final String name;
+    private List<PrometheusColumn> columns;
+    private List<ColumnMetadata> columnsMetadata;
+    private final List<URI> sources;
```

Comment:
Why is this a list? And do we need the URI in the `PrometheusTable` at all?

Reply:
It's a list because some things being connected to by connectors have multiple sources for the data. If this is just weird for all Prometheus setups, we can drop the list?

Reply:
The PrometheusSplitManager uses the sources to the build the access URL now. If it only ever needs to have the one source Prometheus, we could give the SplitManager access directly to the sole config value for Prometheus?

Reply:
Based on your comment sounds like we don't need to list multiple sources.

Reply:
Addressed in ea1b14c7